examples/basic_authentication/routes/users/[id].dart

```diff
@@ -0,0 +1,57 @@
+import 'dart:io';
+
+import 'package:dart_frog/dart_frog.dart';
+import 'package:example/user_repository.dart';
+
+Future<Response> onRequest(RequestContext context, String id) async {
+  return switch (context.request.method) {
+    HttpMethod.get => _getUser(context, id),
+    HttpMethod.patch => _updateUser(context, id),
+    HttpMethod.delete => _deleteUser(context, id),
+    _ => Future.value(Response(statusCode: HttpStatus.methodNotAllowed)),
+  };
+}
+
+Future<Response> _getUser(RequestContext context, String id) async {
+  final user = context.read<User>();
+  if (user.id != id) {
+    return Response(statusCode: HttpStatus.forbidden);
+  }
+  return Response.json(
+    body: {
+      'id': user.id,
+      'name': user.name,
+      'username': user.username,
+    },
+  );
+}
+
+Future<Response> _updateUser(RequestContext context, String id) async {
```

Comment:
Am I missing something or is there no authentication needed to update a user?

Reply:
Not sure I got it, are you asking if the user needs to be authenticated to be updated?

If so, it does and that is already handled by the middleware.

But your comment made me realized that I forgot to check if the authenticated user is the same from the id, so I am gonna address that.

Reply:
> But your comment made me realized that I forgot to check if the authenticated user is the same from the id, so I am gonna address that.

Awesome, that's what I was trying to point out but wasn't completely sure.